File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    int binarySearch(vector<int> &v, int left, int right, int x){
        while (left<=right){
            int mid = left + (right-left)/2;
            if (v[mid] == x) return mid;
            else if (v[mid]>x) right = mid-1;
            else left = mid+1;
        }

        return -1;
    }
    vector<int> twoSum(vector<int>& numbers, int target) {
        for (int i=0; i<numbers.size(); i+=1){
            int tofind = target - numbers[i];
            if (binarySearch(numbers, i+1, numbers.size()-1, tofind) != -1){
                return {i+1, binarySearch(numbers, i+1, numbers.size()-1, tofind)+1};
            }
        }
        return {-1,-1};
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.go
func twoSum(numbers []int, target int) []int {
    for i := range numbers {
        toFind := target - numbers[i]

        idx := binarySearch(numbers, i+1, toFind)

        if idx != -1 {
            return []int{i+1, idx+1}
        }
    }

    return []int{-1,-1}
}
func binarySearch(nums []int, st, target int) int {
    end := len(nums)-1
    for st <= end {
        mid := st + (end - st)/2

        if nums[mid] == target {
            return mid
        } else if nums[mid] > target {
            end = mid - 1
        } else {
            st = mid + 1
        }
    }

    return -1
}